src/components/appTile/AppMenu.js

```diff
@@ -0,0 +1,148 @@
+import build from "../../util/DebugIDUtil";
+import ids from "./ids";
+
+/**
+ *
+ * @author sriram
+ *
+ * Menu items for Apps Tile and table view
+ *
+ */
+const useStyles = makeStyles((theme) => ({
+    menuItem: {
+        padding: 5,
+    },
+    toolbarItemColor: {
+        color: palette.darkBlue,
+    },
+}));
+
+function AppMenu(props) {
+    const {
+        onAppInfoClick,
+        onCommentsClick,
+        onFavoriteClick,
+        baseDebugId,
+        isExternal,
+        isFavorite,
+    } = props;
+    const [anchorEl, setAnchorEl] = useState(null);
+    const open = Boolean(anchorEl);
+    const handleClose = () => setAnchorEl(null);
+    const classes = useStyles();
+    return (
+        <div>
+            <IconButton
+                id={build(baseDebugId, ids.DOT_MENU)}
+                aria-label="More"
+                aria-owns={open ? "long-menu" : null}
+                aria-haspopup="true"
+                onClick={(event) => setAnchorEl(event.currentTarget)}
+                size="small"
+            >
+                <MoreVertIcon />
+            </IconButton>
+            <Menu anchorEl={anchorEl} open={open} onClose={handleClose}>
+                <MenuItem
+                    id={build(baseDebugId, ids.APP_INFO)}
+                    disabled={false}
+                    className={classes.menuItem}
+                    data-disabled={false}
+                    onClick={() => {
+                        onAppInfoClick();
+                        handleClose();
+                    }}
+                >
+                    <InfoIcon className={classes.toolbarItemColor} />
+                    {getMessage("appInfo")}
+                </MenuItem>
+                {!isFavorite && !isExternal && (
+                    <MenuItem
+                        id={build(baseDebugId, ids.FAVORITES)}
+                        disabled={false}
+                        className={classes.menuItem}
+                        data-disabled={false}
+                        onClick={() => {
+                            onFavoriteClick();
+                            handleClose();
+                        }}
+                    >
+                        <FavoriteIcon className={classes.toolbarItemColor} />
+                        {getMessage("addToFavorites")}
+                    </MenuItem>
+                )}
+                {isFavorite && !isExternal && (
+                    <MenuItem
+                        id={build(baseDebugId, ids.FAVORITES)}
+                        disabled={false}
+                        className={classes.menuItem}
+                        data-disabled={false}
+                        onClick={() => {
+                            onFavoriteClick();
+                            handleClose();
+                        }}
+                    >
+                        <UnFavoriteIcon className={classes.toolbarItemColor} />
+                        {getMessage("removeFromFavorites")}
```

Comment:
Could these `ids.FAVORITES` `MenuItem` be combined into 1 by putting the `isFavorite` check as the `MenuItem`'s children?
```jsx

                {!isExternal && (
                    <MenuItem
                        id={build(baseDebugId, ids.FAVORITES)}
                        className={classes.menuItem}
                        onClick={() => {
                            onFavoriteClick();
                            handleClose();
                        }}
                    >{
                        isFavorite
                        ? [<FavoriteIcon className={classes.toolbarItemColor} />,
                            {getMessage("addToFavorites")}]
                        : [<UnFavoriteIcon className={classes.toolbarItemColor} />,
                           {getMessage("removeFromFavorites")}]
                    }</MenuItem>
                )}
```

Reply:
Only slightly different 
` {!isExternal && (
                    <MenuItem
                        id={build(baseDebugId, ids.FAVORITES)}
                        className={classes.menuItem}
                        onClick={() => {
                            onFavoriteClick();
                            handleClose();
                        }}
                    >
                        {isFavorite
                            ? [
                                  <span>
                                      <FavoriteIcon
                                          className={classes.toolbarItemColor}
                                      />
                                      ,{getMessage("addToFavorites")}
                                  </span>,
                              ]
                            : [
                                  <span>
                                      <UnFavoriteIcon
                                          className={classes.toolbarItemColor}
                                      />
                                      ,{getMessage("removeFromFavorites")}
                                  </span>,
                              ]}
                    </MenuItem>`

Reply:
` {!isExternal && (
                    <MenuItem
                        id={build(baseDebugId, ids.FAVORITES)}
                        className={classes.menuItem}
                        onClick={() => {
                            onFavoriteClick();
                            handleClose();
                        }}
                    >
                        {isFavorite
                            ? [
                                  <span>
                                      <FavoriteIcon
                                          className={classes.toolbarItemColor}
                                      />
                                      ,{getMessage("addToFavorites")}
                                  </span>,
                              ]
                            : [
                                  <span>
                                      <UnFavoriteIcon
                                          className={classes.toolbarItemColor}
                                      />
                                      ,{getMessage("removeFromFavorites")}
                                  </span>,
                              ]}
                    </MenuItem>`

Reply:
Btw I didn't knew about `jsx` formatting in comment. Nice! 👍

Reply:
Going back to original code. Adding a `span` throws icon out of alignment.

Reply:
Actually, if you're returning `span` components, then they don't need to be nested in arrays:
```jsx
{ isFavorite ? (<span>...</span>) : (<span>...</span>) }
```
My first example should be the same as your original code, where `FavoriteIcon` and `getMessage` are direct children of the `MenuItem`.

Reply:
So the example code gave a syntax error when I tried. I had to add `<span>` to fix that...

Reply:
Oops, I think I see an error in my first example.
What about removing the braces around the `getMessage` calls?
```jsx
{
    isFavorite
    ? [<FavoriteIcon className={classes.toolbarItemColor} />,
        getMessage("addToFavorites")]
    : [<UnFavoriteIcon className={classes.toolbarItemColor} />,
       getMessage("removeFromFavorites")]
}
```

Reply:
Let me give that a shot once i fix builds!